GitCommands/PathUtil.cs

```diff
@@ -72,20 +72,44 @@ public static bool IsLocalFile(string fileName)
         }
 
         /// <summary>
-        /// A naive way to check whether the given path is a URL by checking
-        /// whether it starts with either 'http', 'ssh' or 'git'.
+        /// Check whether the given path contains an URL by checking
+        /// whether it can be parsed by the URI parsers.
         /// </summary>
+        /// <remarks>
+        /// Uri.IsWellFormedUriString is used instead of Uri.TryCreate as file URIs
+        /// of the form X:\\directory\\filename should not be treated as URLs (doing so causes
+        /// LocalRepositoryManager.AddAsMostRecentAsync to throw a NotSupportedException)
+        /// </remarks>
         /// <param name="path">A path to check.</param>
-        /// <returns><see langword="true"/> if the given path starts with 'http', 'ssh' or 'git'; otherwise <see langword="false"/>.</returns>
+        /// <returns><see langword="true"/> if the given path contains a URL; otherwise <see langword="false"/>.</returns>
         [Pure]
         public static bool IsUrl(string path)
```

Comment:
This method doesn't add much value any more. It's basically just `Uri.IsWellFormedUriString`. Do we still need it?

Reply:
I've removed this method and replaced references to it by calling `Uri.IsWellFormedUriString`.